pkg/audit/cursor.go

```diff
@@ -54,49 +55,30 @@ func (cursor *Cursor) NextStripe(ctx context.Context) (stripe *Stripe, err error
 		return nil, err
 	}
 
-	if len(pointerItems) == 0 {
-		return nil, nil
-	}
-
-	pointerItem, err := getRandomPointer(pointerItems)
-	if err != nil {
-		return nil, err
-	}
-
-	path = pointerItem.Path
-
 	// keep track of last path listed
 	if !more {
 		cursor.lastPath = ""
 	} else {
 		cursor.lastPath = pointerItems[len(pointerItems)-1].Path
 	}
 
-	// get pointer info
-	pointer, err := cursor.pointerdb.Get(path)
-	if err != nil {
-		return nil, err
-	}
+	var pointer *pb.Pointer
+	errGroup := new(errs.Group)
+	for {
```

Comment:
If you had all bad pointers, would this run indefinitely?  Maybe random.Perm() would help?

Reply:
I was also thinking this but it seemed unlikely that the only pointers in the pointerdb all would be segment size 0 and inline? but possible!